runtime/lib/backend/LightningKokkosSimulator.cpp

```diff
@@ -412,17 +444,22 @@ auto LightningKokkosSimulator::PartialCounts(const std::vector<QubitIdType> &wir
         for (auto wire : dev_wires) {
             basisState[idx++] = li_samples[shot * numQubits + wire];
         }
-        counts[basisState.to_ulong()] += 1;
+        counts(static_cast<size_t>(basisState.to_ulong())) += 1;
     }
-
-    return {eigvals, counts};
 }
 
 auto LightningKokkosSimulator::Measure(QubitIdType wire) -> Result
 {
     // get a measurement
     std::vector<QubitIdType> wires = {reinterpret_cast<QubitIdType>(wire)};
-    auto &&probs = this->PartialProbs(wires);
+
+    std::vector<double> probs(1U << wires.size());
+
+    {
+        MemRefT<double, 1> buffer{probs.data(), probs.data(), 0, {probs.size()}, {1}};
+        MemRefView<double, 1> buffer_view(&buffer, probs.size());
```

Comment:
Maybe we could have a constructor that also accepts vectors to avoid this problem of manually having to instantiate MemRef descriptors which the user of the QuantumDevice interface shouldn't need to worry about.

Reply:
MemRefView is just a view to MemRef types and shouldn't create/store a MemRef from a std::vector. I supposed the current implementation isn't smelly. lmk if you think otherwise.

Reply:
The MemRef type is already a non-owning view on the underlying data, copying, instantiating, or deleting MemRef descriptors does in no way affect the underlying memory or data.
The problem this new class should thus solve is to make it more convenient to a C++ implementation to access and iterate over the data (akin to a C++-style view), whereas the MemRef type can be considered more of a lower-level "C-style" view.

Reply:
Thanks for the suggestion! Fixed [170ab09](https://github.com/PennyLaneAI/catalyst/pull/109/commits/170ab09406637677c830a808ab5fe21d57fc34f7)